lib/ecto/changeset.ex

```diff
@@ -1641,6 +1641,15 @@ defmodule Ecto.Changeset do
   defp codepoints_length(<<_, rest::binary>>, acc), do: codepoints_length(rest, acc + 1)
   defp codepoints_length(<<>>, acc), do: acc
 
+  defp list_length(%{types: types}, field, value) do
+    case Map.fetch(types, field) do
```

Comment:
This is going to break in a couple situations:

1. When the first element of the tuple is `:embed`
2. When the user has a custom type that works as a list

So I think we should pattern match rather on this:

```elixir
{:ok, {tag, relation}} when tag in [:embed, :assoc] ->
  ...
_ ->
  length(value)
```